deps/baseconfig/src/typeoperators.h

```diff
@@ -71,27 +72,28 @@ __forceinline ENUMTYPE &operator>>=(ENUMTYPE &a, SizedInteger<ENUMTYPE>::type co
-__forceinline ENUMTYPE operator--(ENUMTYPE const &a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE)(--((SizedInteger<ENUMTYPE>::type &)a)); } \
-__forceinline constexpr ENUMTYPE operator+(ENUMTYPE const a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE)(((SizedInteger<ENUMTYPE>::type)a) + ((SizedInteger<ENUMTYPE>::type)b)); } \
-__forceinline constexpr ENUMTYPE operator-(ENUMTYPE const a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE)(((SizedInteger<ENUMTYPE>::type)a) - ((SizedInteger<ENUMTYPE>::type)b)); } \
-__forceinline constexpr ENUMTYPE operator*(ENUMTYPE const a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE)(((SizedInteger<ENUMTYPE>::type)a) * ((SizedInteger<ENUMTYPE>::type)b)); } \
-__forceinline constexpr ENUMTYPE operator/(ENUMTYPE const a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE)(((SizedInteger<ENUMTYPE>::type)a) / ((SizedInteger<ENUMTYPE>::type)b)); } \
-__forceinline constexpr ENUMTYPE operator%(ENUMTYPE const a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE)(((SizedInteger<ENUMTYPE>::type)a) % ((SizedInteger<ENUMTYPE>::type)b)); } \
-__forceinline ENUMTYPE &operator+=(ENUMTYPE &a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE &)((SizedInteger<ENUMTYPE>::type &)a = ((SizedInteger<ENUMTYPE>::type &)a) + ((SizedInteger<ENUMTYPE>::type)b)); } \
-__forceinline ENUMTYPE &operator-=(ENUMTYPE &a, SizedInteger<ENUMTYPE>::type const b) { return (ENUMTYPE &)((SizedInteger<ENUMTYPE>::type &)a = ((SizedInteger<ENUMTYPE>::type &)a) - ((SizedInteger<ENUMTYPE>::type)b)); } \
+__forceinline ENUMTYPE operator++(ENUMTYPE const &a) { return (ENUMTYPE)(++((UNDERLYING &)a)); } \
+__forceinline ENUMTYPE operator--(ENUMTYPE const &a) { return (ENUMTYPE)(--((UNDERLYING &)a)); } \
+__forceinline constexpr ENUMTYPE operator+(ENUMTYPE const a, ENUMTYPE const b) { return (ENUMTYPE)(((UNDERLYING)a) + ((UNDERLYING)b)); } \
+__forceinline constexpr ENUMTYPE operator-(ENUMTYPE const a, ENUMTYPE const b) { return (ENUMTYPE)(((UNDERLYING)a) - ((UNDERLYING)b)); } \
+__forceinline constexpr ENUMTYPE operator*(ENUMTYPE const a, ENUMTYPE const b) { return (ENUMTYPE)(((UNDERLYING)a) * ((UNDERLYING)b)); } \
+__forceinline constexpr ENUMTYPE operator/(ENUMTYPE const a, ENUMTYPE const b) { return (ENUMTYPE)(((UNDERLYING)a) / ((UNDERLYING)b)); } \
+__forceinline constexpr ENUMTYPE operator%(ENUMTYPE const a, ENUMTYPE const b) { return (ENUMTYPE)(((UNDERLYING)a) % ((UNDERLYING)b)); } \
+__forceinline ENUMTYPE &operator+=(ENUMTYPE &a, ENUMTYPE const b) { return (ENUMTYPE &)((UNDERLYING &)a = ((UNDERLYING &)a) + ((ENUMTYPE)b)); } \
+__forceinline ENUMTYPE &operator-=(ENUMTYPE &a, ENUMTYPE const b) { return (ENUMTYPE &)((UNDERLYING &)a = ((UNDERLYING &)a) - ((ENUMTYPE)b)); } \
+__forceinline ENUMTYPE operator++(ENUMTYPE const &a, int) { return (ENUMTYPE)(((UNDERLYING &)a)++); } \
+__forceinline ENUMTYPE operator--(ENUMTYPE const &a, int) { return (ENUMTYPE)(((UNDERLYING &)a)--); } \
```

Comment:
This is post-increment. It was implemented as pre-increment before and was therefore wrong.